src/redis-cli.c

```diff
@@ -7332,8 +7370,8 @@ static redisReply *sendScan(unsigned long long *it) {
     redisReply *reply;
 
     if (config.pattern)
-        reply = redisCommand(context,"SCAN %llu MATCH %s",
-            *it,config.pattern);
+        reply = redisCommand(context, "SCAN %llu MATCH %b",
+            *it, config.pattern, sdslen(config.pattern));
```

Comment:
are you sue you can use `redisCommand` this way with binary strings?
i see i recently did something different to support binary strings in `--bigkeys`:
```c
        const char* argv[] = {"TYPE", keys->element[i]->str};
        size_t lens[] = {4, keys->element[i]->len};
        redisAppendCommandArgv(context, 2, argv, lens);
```
i guess i wasn't familiar with `%b`